subsys/logging/log_core.c

```diff
@@ -298,7 +305,7 @@ void log_core_init(void)
 
 	/* Set default timestamp. */
 	timestamp_func = timestamp_get;
-	log_output_timestamp_freq_set(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+	log_output_timestamp_freq_set(freq);
```

Comment:
Couldn't we actually make this a lot simpler by not requiring a separate static function at all:

```
if (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC > 1000000)
        timestamp_func = k_uptime_get_32;
        log_output_timestamp_freq_set(MSEC_PER_SEC);
} else {
        timestamp_func = k_cycle_get_32;
        log_output_timestamp_freq_set(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
}
```

I'll leave it up to you to decide if it makes sense/is worth it.